### ISSUE TITLE
Clone chds non-functional (area51t)

### ISSUE THREAD

**Issue by Sanaki** [mamedev/mame#9409]
Tested using MAME 0.241 on x86_64 Linux, using the area51t chd as a clone. When loading a game with a clone chd rather than a complete one, the game loads fine, no error regarding the parent is thrown (even if the parent is removed), but the data from the parent can't be accessed and results in corruption, black screens, etc. To my understanding, clone support should be present and fully functional for any game that has a parent.

Commands used to generate clone:
```
chdman extracthd -i area51t.chd -o area51t.img
<move/remove original chd>
chdman createhd -i area51t.img -op area51.chd -o area51t.chd
```
Since it's a clone, chs values are automatic (and do match), the data and full SHA1 values match the non-clone, everything looks fine. I confirmed the full chd loads fine as well. Am I missing something obvious here?

**Comment by cuavas** [comment]
You mean delta CHDs?  MAME doesn’t automatically search for the base CHD if it encounters a delta CHD.  There was a PR that was supposed to scan the directory containing the CHD for possible base CHDs, but it leaked objects (because memory management is hard, apparently).  The CHD class expects the base CHD file to be supplied when first opening the CHD, hence you need to know that you’re expecting a delta CHD and what the correct base CHD is *before* attempting to open the CHD.  To be able to work well with the way MAME searches for assets, it would need to be changed around to have a callback saying, “Can you find a CHD with this digest?” when it encounters a delta CHD so MAME could do the parent walk at that point to guess possible names/locations.

**Comment by Sanaki** [comment]
Ahhh. That explains it. And yes, that's exactly what I was referring to. Are there any delta chds currently functional in MAME then, or is it just functionality awaiting future implementation?

**Comment by i30817** [comment]
I believe libchd/chd-rs also doesn't do this, and thus retroarch cores also fails at this. I believe that 'by default' both should search the current directory (with the option to use other strategies), so interested people should maybe open bugs about this in those libraries, or the respective libretro cores if the libraries reject the idea.

But if upstream adopts a 'by default, the current directory is searched for parents' i suspect the libraries would follow.

BTW, does chd here has the extension that was accepted in libchd (i think) to allow multiple levels of 'parents' to a final delta chd?

**Comment by cuavas** [comment]
This has been addressed by d1172bf710f2a7b1777ed7687f5ba65748d90ee9 (will be in MAME 0.260).  A year and a half turnaround isn’t great, but we got there.
